conans/client/remote_manager.py

```diff
@@ -61,12 +61,17 @@ def upload_recipe(self, ref, remote, retry, retry_wait, policy, remote_manifest)
         if policy == UPLOAD_POLICY_SKIP:
             return ref
 
-        ret, rev_time = self._call_remote(remote, "upload_recipe", ref,
-                                          the_files, retry, retry_wait, policy, remote_manifest)
+        ret = self._call_remote(remote, "upload_recipe", ref, the_files, retry, retry_wait,
+                                policy, remote_manifest)
 
         # Update package revision with the rev_time (Created locally but with rev_time None)
         with self._cache.package_layout(ref).update_metadata() as metadata:
-            metadata.recipe.time = rev_time
+            if metadata.recipe.time is None:
+                _, _, rev_time = self._call_remote(remote, "get_recipe_snapshot", ref)
```

Comment:
Sounds like calling this API might be a bit overkill for just getting the revision time.
I might need an API call to just get the PREF of a given REF#RREV#BID, for the exact binary ID resolution.

Reply:
We have to talk about this. Ping me @memsharded this morning.